webui/react/src/pages/ExperimentDetails/ExperimentMultiTrialTabs.tsx

```diff
@@ -165,6 +165,7 @@ const ExperimentMultiTrialTabs: React.FC<Props> = ({
           disableTitle
           notes={{ contents: experiment.notes ?? '', name: 'Notes' }}
           onError={handleError}
+          onPageUnloadHook={unstable_useBlocker}
```

Comment:
q: What does this do?

Reply:
the `Note` component that `NoteCards` renders should prompt the user when leaving the page if they're sure they want to leave if the note in question has dirty state. Before the split, `Note` imported `unstable_useBlocker` directly from `react-router-dom` to do the prompt, but that requires a data router in the component context, so now we pass the hook in from the caller via `onPageUnloadHook`.